scss/_tables.scss

```diff
@@ -1,31 +1,42 @@
+// stylelint-disable selector-max-universal
 //
 // Basic Bootstrap table
 //
 
 .table {
+  --table-cell-padding: #{$table-cell-padding};
+  --table-border-width: 0;
+  --table-border-bottom-width: #{$table-border-width};
+  --table-bg: #{$table-bg};
+  --table-accent-bg: #{$table-accent-bg};
+  --table-hover-bg: #{$table-hover-bg};
+  --table-hover-color: #{$table-hover-color};
+  --table-active-bg: #{$table-active-bg};
+  --table-children-border-width: #{$table-children-border-width};
+
   width: 100%;
   margin-bottom: $spacer;
   color: $table-color;
   vertical-align: $table-cell-vertical-align;
-  background-color: $table-bg; // Reset for nesting within parents with `background-color`.
+  border-color: $table-border-color;
 
-  th,
-  td {
-    padding: $table-cell-padding;
-    border-bottom: $table-border-width solid $table-border-color;
+  // th & td
+  > * > * > * {
+    padding: var(--table-cell-padding);
+    background-color: var(--table-bg);
+    border-width: var(--table-border-width) var(--table-border-width) var(--table-border-bottom-width);
   }
 
-  tbody {
+  > tbody {
     vertical-align: inherit;
   }
 
-  thead th {
+  > thead {
     vertical-align: bottom;
-    border-bottom-color: $table-head-border-color;
   }
 
-  tbody + tbody {
-    border-top: (2 * $table-border-width) solid $table-border-color;
+  > :not(caption) + * {
+    border-top-width: var(--table-children-border-width);
   }
```

Comment:
Are those direct children selector necessary, since they apply to every `.table`? 

It always applies, even in nested `.table`s — except when nesting a `<table>` without `.table` in a `.table`. Is that your goal?

The same goes for `> * > * > *` since all values are set with custom properties: they can be easily overriden in variants by setting those custom properties directly (which is already what you're doing).

Am I missing something?

Reply:
Styles can be leaked to `<table>`s without `.table` class 😉

Reply:
If we're trying to solve this case, shouldn't we show an example of a `.table` containing a standard `<table>`?

It looks odd to me that one could use our table styles for a parent table and not in its table children :D

Reply:
This seems to be a common case: https://codepen.io/MartijnCuppens/pen/wvaRaVP

This PR version: https://codepen.io/MartijnCuppens/pen/NWqeGWY

Reply:
Feels weird to me, I'd recommend to move the modal outside. But if you experienced issues with this kind of things, it probably should be addressed in some way :+1: